src/core/scene/a-scene.js

```diff
@@ -169,7 +169,7 @@ var AScene = module.exports = registerElement('a-scene', {
         if (!camera || !canvas) { return; }
 
         // Update canvas.
-        if (!isMobile) {
+        if (!isMobile && !canvas.provided) {
```

Comment:
yeah, it should be a data attribute IMO. I've seen non-browser properties get unset; tbh I don't know how browsers always handle those.